src/osd/OSD.cc

```diff
@@ -2403,6 +2403,21 @@ OSD::OSD(CephContext *cct_,
   trace_endpoint.copy_name(ss.str());
 #endif
 
+  // Determine scheduler type for this OSD
+  auto get_scheduler_type = [&conf = cct->_conf]() {
+    const std::string _type = conf.get_val<std::string>("osd_op_queue");
+    const std::string *type = &_type;
+    if (*type == "debug_random") {
+      static const std::vector<std::string> index_lookup = { "mclock_scheduler",
+                                                             "wpq" };
+      std::mt19937 random_gen(std::random_device{}());
+      auto which = random_gen() % index_lookup.size();
+      type = &index_lookup[which];
+    }
+    return *type;
+  };
+  osd_op_queue = get_scheduler_type();
```

Comment:
Instead of making this an OSD member, pass it by the OSDShard constructor and update users of osd_op_queue to instead invoke a get_type()  method on the queue itself.

Reply:
I have made the necessary changes. Please review.